src/test/kotlin/com/ampnet/identityservice/persistence/repository/FaucetTaskRepositoryTest.kt

```diff
@@ -0,0 +1,116 @@
+package com.ampnet.identityservice.persistence.repository
+
+import com.ampnet.identityservice.TestBase
+import com.ampnet.identityservice.persistence.model.FaucetTaskStatus
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import java.time.ZonedDateTime
+import java.util.UUID
+
+@DataJpaTest
+@ExtendWith(value = [SpringExtension::class])
+@AutoConfigureTestDatabase
+class FaucetTaskRepositoryTest : TestBase() {
+
+    @Autowired
+    private lateinit var faucetTaskRepository: FaucetTaskRepository
+
+    @Test
+    fun mustCorrectlyAddAndFlushAddressQueue() {
+        val chain1Addresses = listOf("addr1", "addr2", "addr3", "addr1", "addr4", "addr2")
+        val chain2Addresses = listOf("addr1", "addr4", "addr5")
+
+        suppose("some addresses are added to the queue") {
+            chain1Addresses.forEach { faucetTaskRepository.addAddressToQueue(it, 1L) }
+            chain2Addresses.forEach { faucetTaskRepository.addAddressToQueue(it, 2L) }
+        }
+
+        val task1Uuid = UUID.randomUUID()
+        val now = ZonedDateTime.now()
+
+        suppose("address queue is flushed for chain 1") {
+            faucetTaskRepository.flushAddressQueueForChainId(task1Uuid, 1L, now)
+        }
+
+        val uniqueChain1Addresses = chain1Addresses.toSet()
+
+        verify("task is created for flushed addresses for chain 1") {
+            val task = faucetTaskRepository.findById(task1Uuid)
+            assertThat(task).hasValueSatisfying {
```

Comment:
Nice, I didn't about this.

Reply:
Yeah, AssertJ has some really nice assertions 😄